### PR TITLE
fix(editor): Handling router errors when navigation is canceled by user

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/editor-ui/src/components/WorkflowShareModal.ee.vue [hunk 1]
```diff
@@ -155,6 +155,7 @@ import { useWorkflowsEEStore } from '@/stores/workflows.ee';
 import { ITelemetryTrackProperties } from 'n8n-workflow';
 import { useUsageStore } from '@/stores/usage';
 import { BaseTextKey } from '@/plugins/i18n';
+import { isNavigationFailure } from 'vue-router';
```
Comment (ivov) [inline packages/editor-ui/src/components/WorkflowShareModal.ee.vue:158]: For learning, we cannot [set it up like in the docs](https://router.vuejs.org/api/#isnavigationfailure)?

File: packages/editor-ui/src/components/MainSidebar.vue [hunk 2]
```diff
@@ -419,55 +412,13 @@ export default mixins(
...
-						type: 'success',
-					});
-				} else if (confirmModal === MODAL_CLOSE) {
-					return;
-				}
-			} else {
-				if (this.$router.currentRoute.name !== VIEWS.NEW_WORKFLOW) {
-					this.workflowsStore.setWorkflowId(PLACEHOLDER_EMPTY_WORKFLOW_ID);
-					this.$router.push({ name: VIEWS.NEW_WORKFLOW });
+		goToRoute(route: string | { name: string }) {
```
Comment (ivov) [inline packages/editor-ui/src/components/MainSidebar.vue:415]: `createNewWorkflow` was an unused leftover?

**Review by ivov — APPROVED** [review]
Working as expected, just two questions for understanding.

File: packages/editor-ui/src/components/WorkflowShareModal.ee.vue (hunk 1 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/WorkflowShareModal.ee.vue:158]: From what I've been able to test out, this hook is not triggered if navigation is canceled from another guard (such as `beforeRouteLeave`).

File: packages/editor-ui/src/components/MainSidebar.vue (hunk 2 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/MainSidebar.vue:415]: Yep, that's when we had _New workflow_ menu item in the sidebar (before the _Workflows_ page).

**Pull request merged**